jsparagus/parse_table.py

```diff
@@ -1477,6 +1479,73 @@ def transform() -> typing.Iterator[None]:
 
         consume(transform(), progress)
 
+    def fold_replay_unwind(self, verbose: bool, progress: bool) -> None:
+        """Convert Replay action falling into Unwind action to an Unwind action which
+        replay less terms."""
+        if verbose or progress:
+            print("Fold Replay followed by Unwind actions.")
+
+        maybe_unreachable_set: OrderedSet[StateId] = OrderedSet()
+
+        def try_transform(s: StateAndTransitions) -> bool:
+            if len(s.epsilon) != 1:
+                return False
+            replay_term, replay_dest_idx = next(iter(s.epsilon))
+            if not isinstance(replay_term, Replay):
+                return False
+            replay_dest = self.states[replay_dest_idx]
+            if len(replay_dest.epsilon) != 1:
+                return False
+            unwind_term, unwind_dest_idx = next(iter(replay_dest.epsilon))
+            if not unwind_term.update_stack():
+                return False
+            stack_diff = unwind_term.update_stack_with()
+            if stack_diff.replay <= 0:
+                return False
```

Comment:
can this condition be moved to `StackDiff` method with some descriptive name, or add some comment here?

to my understanding, `replay` field cannot be negative for `Unwind` action (according to the comment above `replay` field).
and 0 is possible but it's excluded because we cannot apply any transform if there's no traffic, right?
or, if this expects not-`Unwind` action, this condition really needs some explicit name

Reply:
At this stage `stack_diff.replay` cannot be negative for `Unwind` but might be in the upcoming patches.
0 is indeed excluded because we have no optimization opportunity.